### ISSUE TITLE
Request .mp3 BEFORE .mp3.bz2 over FastDL?

### ISSUE THREAD

**Issue by stickz** [Facepunch/garrysmod-issues#2753]
The mp3 audio format already contains compression. The size remains almost constant when adding a second .bz2 layer. Requesting it in .bz2 form first increases download times. Extra resources are spent decompressing it. Some server operators only include .mp3 on their fastdl as skipping .mp3.bz2 is faster.

Would it make sense to modify the source-engine to request .mp3 BEFORE .mp3.bz2 over FastDL? This would increase download times for game clients; by skipping the decompression stage all-together.

**Comment by Kefta** [comment]
Almost all servers compress all files, though, so now you're adding an extra request

**Comment by thegrb93** [comment]
How much time does it save? I imagine the 404 and direct download from the server would take much longer than just using bz2.

**Comment by stickz** [comment]
bz2 takes ages to decompress as it's single threaded. A 404 is significantly faster at 100ms than decompression; especially when file sizes are larger and produce no reduction in size. (download times)

**Comment by TehBigA** [comment]
Yeah Source retries quite a bit before it gives up. I've seen this issue in CS:GO recently when servers' fastdl starts throwing 500s.
